diesel_tests/tests/update.rs

```diff
@@ -257,3 +259,22 @@ fn struct_with_option_fields_treated_as_null() {
     assert_eq!(Ok(&expected_post), updated_post.as_ref());
     assert_eq!(Ok(&expected_post), post_in_database.as_ref());
 }
+
+#[test]
+fn update_with_no_changes() {
+    #[derive(AsChangeset)]
+    #[table_name="users"]
+    struct Changes {
+        name: Option<String>,
+        hair_color: Option<String>,
+    }
+
+    let connection = connection();
+    let changes = Changes { name: None, hair_color: None, };
+    let updated_rows = update(users::table).set(&changes).execute(&connection);
+
+    match updated_rows {
+        Ok(_) => panic!("noop update was expected to return an error"),
+        Err(e) => assert!(e.description().starts_with("There are no changes to save.")),
+    }
```

Comment:
You can replace this match with an `unwrap` and adding `#[should_panic(expected = "There are no changes to save.")]` to the fn